apps/andi/lib/andi/migration/migrations.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Andi.Migration.Migrations do
+  @moduledoc """
+  Contains all migrations that run during bootup.
+  """
+  use GenServer, restart: :transient
+
+  import Andi, only: [instance_name: 0]
+
+  @instance Andi.instance_name()
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  def init(_args) do
+    migrate_modified_dates()
+
+    {:ok, :ok, {:continue, :stop}}
+  end
+
+  def handle_continue(:stop, state) do
+    {:stop, :normal, state}
+  end
+
+  def migrate_modified_dates do
+    if is_nil(Brook.get!(@instance, :migration, "modified_date_migration_completed")) do
```

Comment:
`if` block seems like a smell to me. Could you move the `Brook.get!` call to `init` and pipe it to `migrate_modified_dates` to guard on `nil`?

```elixir
def inti(_) do
  Brook.get!(@instance, :migration, "modified_date_migration_completed")
  |> migrate_modified_dates()

  {:ok, :ok, {:continue, :stop}}
end

defp migrate_modified_dates(dates) when is_nil(dates) do
  Brook.Event.send(@instance, "migration:modified_dates", :andi, %{})
end

defp migrate_modified_dates(_), do: nil
```

Reply:
I'd like init to be just a list of migration function calls so you can scan the init and know what all migrations are being run without having to understand details.
```
 def init(_args) do
    migrate_modified_dates()
    migrate_skyrim_mods()
    migrate_friends()

    {:ok, :ok, {:continue, :stop}}
  end
```
The counter proposal I'd make is below, though I think I'd just prefer the if at that point.
```
def migrate_modified_dates do
    Brook.get!(@instance, :migration, "modified_date_migration_completed")
    |> migrate_modified_dates()
  end

  defp migrate_modified_dates(nil), do: Brook.Event.send(@instance, "migration:modified_dates", :andi, %{})

  defp migrate_modified_dates(_), do: nil
```